PSO Speed up

NicerNewerCar opened the pull request:
* Closes #218 
* Initial results are great, cutting the optimization time to 5/6 seconds [Debug] 2/3 seconds [Release] on my machine
* Currently, I am validating the accuracy of this. I've noticed sometimes it likes to optimize to a nonideal position (ie. not in the correct position on the radiograph). Not sure if this is an issue with my changes or something pre-existing.
    * @amymmorton @jcfr If you both could play around with this PR, that may help expedite the validation process.

jcfr commented on libautoscoper/src/PSO.hpp:
```diff
@@ -12,12 +12,37 @@
+  Particle() {
+    ncc_val = -1.0f;
+  }
+  Particle(std::vector<float> pos) {
```
```suggestion
  Particle(const std::vector<float>& pos) {
```

jcfr reviewed (commented):
Moving in that direction makes sense :ok_hand: 

Could you contribute the API changes (introduction of `Particle` and vector of `Particle`) into its own pull request.

That way the pull request introducing the threading support will be reduce to its strict minimum and the review will be easier ?

NicerNewerCar commented:
See #223 

Rebase this branch once that is integrated.

NicerNewerCar commented:
Rebased this with #223, Seems some of the errors I corrected over there led to this not being as fast. 223 Runs in 3-5 seconds while adding the multithreading added enough overhead that it actually slowed this down to 8-9 seconds.
